features/storage/FEATURE_STORAGE/cfstore/source/cfstore.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+ * mbed Microcontroller Library
+ * Copyright (c) 2006-2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/** @file  cfstore.cpp
+ *
+ * C++ Interface to Configuration Store Implementation.
+ *
+ */
+
+
+#include "cfstore_config.h"
+#include "cfstore_debug.h"
+#include "cfstore_os.h"
+#include "cfstore.h"
+#include <cstdio>
+#include <stdio.h>
+#include <stdarg.h>
+
+namespace cfstore {
+
+#define CFSTORE_SEMAPHORE_SLOTS  0
+
+
+int32_t Cfstore::initialised_count = 0;
+cfstore_driver_t *Cfstore::driver;
+cfstore_capabilities_t Cfstore::caps;
+cfstore_driver_version_t Cfstore::version;
+Mutex Cfstore::mtx_lock;
```

Comment:
Do these get garbage collected if not used? I think it will require the singleton_ptr class